src/engines/collaborative_filtering.py

```diff
@@ -0,0 +1,55 @@
+from src.utils import db, sc
+
+from .engine import Engine
+
+import pandas as pd
+from sqlalchemy import text
+
+from pyspark.sql import SQLContext
+from pyspark.ml.feature import StringIndexer
+from pyspark.ml.recommendation import ALS
+
+class CollaborativeFiltering(Engine):
+
+    __engine_priority__ = 5
+
+    def train(self):
+
+        for media in self.__media__ :
+
+            sqlContext = SQLContext(sc)
+
+            df = media.get_ratings()
+
+            sparkDF = sqlContext.createDataFrame(df)
+
+            itemIdName = media.id
+
+            if media.tablename_media == "book" :
+                itemIdName = "isbn_id"
+                stringIndexer = StringIndexer(inputCol="isbn", outputCol="isbn_id")
+                model = stringIndexer.fit(sparkDF)
+                sparkDF = model.transform(sparkDF)
+            
+            als = ALS(userCol="user_id", itemCol=itemIdName, ratingCol="rating", coldStartStrategy="drop")
+            model = als.fit(sparkDF)   
+
+            usersAsDF = pd.read_sql_query("SELECT user_id FROM public.user WHERE password_hash!='no_pwd'", con=engine)
+            userList = [ [u] for u in usersAsDF['user_id'].tolist() ]
+
+            nbElem = 10
+
+            modelGest = model.recommendForUserSubset(sqlContext.createDataFrame([userList], schema=['user_id']), nbElem)
```

Comment:
`[userList]` format of this is: `[ [ [0], [1], [2] ] ]` with 0, 1 and 2 as user_id, it is normal ?